### PR TITLE
Firecracker v1.0.2

### PR DESCRIPTION
**Pull request by andreitraistaru**
## Reason

Linux-loader v0.4.0 has been yanked because of the cmdline vulnerability, we need to upgrade the linux-loader crate in order to fix the vulnerability. This is done by forking linux-loader v0.7.0 into Firecracker repo and consuming it directly.

Furthermore, the PR contains commits backported from main in order to fix different issues solved meanwhile.

## License Acceptance

By submitting this pull request, I confirm that my contribution is made under
the terms of the Apache 2.0 license.

## PR Checklist

- [ ] All commits in this PR are signed (`git commit -s`).
- [ ] If a specific issue led to this PR, this PR closes the issue.
- [ ] The description of changes is clear and encompassing.
- [ ] Any required documentation changes (code and docs) are included in this PR.
- [ ] New `unsafe` code is documented.
- [ ] API changes follow the [Runbook for Firecracker API changes][2].
- [ ] User-facing changes are mentioned in `CHANGELOG.md`.
- [ ] All added/changed functionality is tested.

---

- [ ] This functionality can be added in [`rust-vmm`][1].

[1]: https://github.com/rust-vmm
[2]: https://github.com/firecracker-microvm/firecracker/blob/main/docs/api-change-runbook.md

### CODE REVIEW

**Review by mattschlebusch — COMMENTED** [review]
I know the PR is still in draft, quick thoughts at a glance:

* Do we need to fork the linux-loader package? The reasoning here needs to be documented in comments at the least I think.
* With regards to the `.bin` files, they certainly bloat the repo - Ideally all files in a code repository should be human-readable and differences between versions viewable in some sort of differential comparison. Here are some options to consider:
  * If the .bin files are intended for Firecracker users to consume and use in a production environment, they should be in the `resources` folder in the root of the workspace.
  * If they are needed for integration or unit tests they could be in a `resources/tests` folder for all tests in the workspace to use.
  * If there is any possibility to programmatically generate these binary files at build time, we should do so. Currently Firecracker generates seccomp binary files in [vmm/build.rs](https://github.com/firecracker-microvm/firecracker/blob/firecracker-v1.0/src/vmm/build.rs). While this file will bloat before it gets refactored out, is the correct entry for this type work in my opinion.

**Comment by aghecenco** [comment]
>   * If they are needed for integration or unit tests they could be in a `resources/tests` folder for all tests in the workspace to use.
>   * If there is any possibility to programmatically generate these binary files at build time, we should do so. 

I would go with the first option (move the bins and update the paths) in order to avoid adding (potentially loads of) extra infrastructure just to generate the test images for code that we're temporarily pulling in for a short lived maintenance. The instructions on how to generate them have been "imported" along with the crate (with [this doc](https://github.com/rust-vmm/linux-loader/blob/main/docs/TESTING.md)).

To be extra pedantic, we could remove the parts of linux-loader that we don't care about in Firecracker from the Firecracker codebase (i.e. PVH stuff, bench tests that don't get run anywhere).

File: src/linux_loader/Cargo.toml [hunk 1]
```diff
@@ -24,10 +25,6 @@ vm-memory = "0.9.0"
 criterion = "0.3.5"
 vm-memory = { version = "0.9.0", features = ["backend-mmap"] }
 
-[[bench]]
```
[the remark this answers is not in the document]
Comment (andreitraistaru) [reply src/linux_loader/Cargo.toml]: Done. Thanks for advice!

**Comment by ShadowCurse** [comment]
Btw, linux-loader 0.8 was released yesterday so maybe update the pulled version as well. Well the only update there was update of vm-memory from 0.9.0 to 0.10.0.

Also it would be nice to update PR description to reflect other changes such as modifications of baselines in benchmarks and so on.

**Comment by roypat** [comment]
> Btw, linux-loader 0.8 was released yesterday so maybe update the pulled version as well. Well the only update there was update of vm-memory from 0.9.0 to 0.10.0.
> 
> Also it would be nice to update PR description to reflect other changes such as modifications of baselines in benchmarks and so on.

I am slowly working on updating all of our rust-vmm dependencies. Due to https://github.com/rust-vmm/community/issues/136, they all have to be updated together. I'm hoping to get a PR for that going this week, as a companion to https://github.com/firecracker-microvm/firecracker/pull/3248

**Review by luminitavoicu — APPROVED** [review]
Looking good! I only have a nit.

File: tests/framework/utils.py [hunk 2]
```diff
@@ -320,7 +320,7 @@ def search_output_from_cmd(cmd: str,
 
 
 def get_files_from(find_path: str, pattern: str, exclude_names: list = None,
-                   recursive: bool = True):
+                   exclude_subdirs: list = None, recursive: bool = True):
```
[the remark this answers is not in the document]
Comment (andreitraistaru) [reply tests/framework/utils.py:323]: Ok, renaming it as suggested. Thanks!

[the remark this answers is not in the document]
Comment (luminitavoicu) [reply tests/framework/utils.py:323]: Sorry, it was a typo, I meant to write `exclude_subdirs`. Just adding the parameter description to the function doc would suffice.

[the remark this answers is not in the document]
Comment (andreitraistaru) [reply tests/framework/utils.py:323]: Solved

**Review by luminitavoicu — APPROVED** [review]

**Review by luminitavoicu — APPROVED** [review]

**Review by aghecenco — APPROVED** [review]

**Pull request merged**